### PR TITLE
Display <none> for kubectl describe secret when type is empty.

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
when describe secret, if its type is null, it is better to show <none>

**Special notes for your reviewer**:
Refer to #45788
**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhangxiaoyu-zidif** [comment]
ok to test

**Comment by cblecker** [comment]
@k8s-bot ok to test

**Comment by zhangxiaoyu-zidif** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by zhangxiaoyu-zidif** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

File: pkg/printers/internalversion/describe.go [hunk 1]
```diff
@@ -1709,7 +1709,11 @@ func describeSecret(secret *api.Secret) (string, error) {
 		skipAnnotations := sets.NewString(api.LastAppliedConfigAnnotation)
 		printAnnotationsMultilineWithFilter(w, "Annotations", secret.Annotations, skipAnnotations)
 
-		w.Write(LEVEL_0, "\nType:\t%s\n", secret.Type)
+		if secret.Type == "" {
```
Comment (deads2k) [inline pkg/printers/internalversion/describe.go]: our defaulting sets this to opaque.  How did you get an empty?  If you can't get an empty via deserialization, I'd rather tighten our validator.

@liggitt I'm assuming the defaulting and validation was you?

Comment (liggitt) [inline pkg/printers/internalversion/describe.go]: I don't think I added secret type defaulting, but it's hard to remember

**Comment by deads2k** [comment]
Looks like it was clayton:
https://github.com/kubernetes/kubernetes/blame/master/pkg/api/v1/defaults.go#L210

**Comment by deads2k** [comment]
Or rather
https://github.com/kubernetes/kubernetes/blame/release-1.0/pkg/api/v1beta3/defaults.go#L115
.  Looks like we've had it since the beginning.  I'd vote for just
tightening validation.

**Comment by zhangxiaoyu-zidif** [comment]
@deads2k  Thank you for your review. I was confused with pkg/api/types.go. In that file, the comment says Type SecretType is optional. I guess "optional" may be mean "empty"
If we have tighten validation already. It's better to close this PR. Thanks all!

**Pull request closed without merging**